starter-code/main.js

```diff
@@ -54,8 +55,192 @@ mongoClient.connect(`mongodb://localhost:27017/crunchbase`, (error, db) => {
             break;
 
           // Code here next cases!
+
+          case "4":
+            // 4.- List by name all companies founded in february of 2004.
+            db.collection('companies').find({ $and: [{ founded_year: 2004 }, { founded_month: 2 }] }, { name: 1, _id: 0 }).toArray((error, result) => {
+              if (error) {
+                console.log(error);
+                rl.question(`\nType enter to continue: `, (answer) => { mainMenu() });
+              } else {
+                console.log(result);
+                rl.question(`\nType enter to continue: `, (answer) => { mainMenu() });
+              }
+            })
+            break;
+
+          case "5":
+            // 5.- List by name all companies founded in the summer of 2004 (april to june) sorted by date.
+            db.collection('companies').find({ $and: [{ founded_year: 2004 }, { founded_month: { $lte: 6 , $gte: 4} }] }, { name: 1, _id: 0 }).sort({ founded_day: 1, founded_month: 1 }).toArray((error, result) => {
```

Comment:
```suggestion
find( { $and: [{founded_year: 2004}, { founded_month: {$gt:3} }, { founded_month: {$lt: 7}}] }, { name: 1, _id: 0 }).sort({ created_at: 1 })
```

- Esta diferencia entre MongoCompass y el ejercicio ejecutado en Node os ha pasado a bastantes, y es resultado de que algunas empresas de la base de datos tienen un mismo año, mismo mes, `null` en día y lo ordena de forma diferente. Para evitarlo, podemos usar directamente la propiedad `created_at` para comparar, ya que todas las empresas tienen el mismo formato 😄